### PR TITLE
draft: Avoid semicolon-handle logic

### PR DESCRIPTION
**Pull request by hoshinotsuyoshi**

### CODE REVIEW

File: frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts [hunk 1]
```diff
@@ -1,38 +1,101 @@
...
+          if (i + currentChunkSize > lines.length) {
+            processErrors.push(...errors)
+            break
+          }
+          // NOTE: Prevent excessive memory usage. If currentChunkSize exceeds twice the original chunkSize, return an error.
+          //       The factor of 2 is arbitrary and can be adjusted in the future if necessary.
+          if (currentChunkSize > chunkSize * 2) {
+            processErrors.push(...errors)
+            break
+          }
```
Comment (hoshinotsuyoshi) [inline frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts:62]: It's a bit complex! 🙏 

- On error, we first try parsing by decreasing `currentChunkSize` one line at a time.  
- If that fails, we switch direction: resetting `currentChunkSize` to the default (500) and then increasing it one line at a time.  
  - If it reaches 1000 lines or the end of the file, we give up and terminate the process.  

If this level of complexity is acceptable, I'll add tests for this behavior since there are none yet. What do you think?

Comment (MH4GF) [reply frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts:62]: Basically, it looked fine, but I couldn't figure out what use cases could be solved by descreasing and what use cases could be solved by increasing. 
It would be great if you could write a test, as it should be easier to understand!

I am wondering if when descreasing, it does not duplicate and parse existing parsed code.

**Review by MH4GF — COMMENTED** [review]
Thanks for solving a difficult problem!

File: frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts (hunk 1 above)
Comment (hoshinotsuyoshi) [reply frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts:62]: Thanks for reviewing it! That makes sense—I’ll write some tests to clarify. Please give me a bit of time.  

> I am wondering if when decreasing, it does not duplicate and parse existing parsed code.  

At this point, I believe that shouldn’t happen due to the nature of the parser, but I’ll include some details on that as well. Stay tuned! 🚀

File: frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts [hunk 2]
```diff
@@ -1,38 +1,101 @@
...
+  let currentChunkSize = 0
+  const processErrors: ProcessError[] = []
+
+  for (let i = 0; i < lines.length; ) {
+    if (processErrors.length > 0) break
+    currentChunkSize = chunkSize
+    enum RetryDirection {
+      Decrease = -1, // Shrinking mode
+      Increase = 1, // Expanding mode
     }
```
[the remark this answers is not in the document]
Comment (hoshinotsuyoshi) [reply frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts:28]: Oh, I’m not too concerned about that! 🙏  
I’ll update it to use that approach.

[the remark this answers is not in the document]
Comment (hoshinotsuyoshi) [reply frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts]: updated https://github.com/liam-hq/liam/pull/699/commits/fa395d03f25aad242354e24471dc99de9d6ef937

File: frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts (hunk 1 above)
Comment (hoshinotsuyoshi) [reply frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.ts]: test added!

- 70c4d5cd711d492a8aa157417327fd7fe3130f6c
- ef96e133f384e510efc4da1131a6abc7b6411de8

File: frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.test.ts [hunk 3]
```diff
@@ -41,24 +42,126 @@ describe(processSQLInChunks, () => {
...
+        [
+          'CREATE TABLE t1 (\n  c1 int,\n  c2 int,\n  c3 int,\n  c4 int',
+          [57, null, [error]],
+        ],
+        // 9th: Expanding mode: Tries lines 3-8 (complete statement).
+        [
+          'CREATE TABLE t1 (\n  c1 int,\n  c2 int,\n  c3 int,\n  c4 int\n);',
+          [null, null, []],
+        ],
+      ]
```
Comment (MH4GF) [inline frontend/packages/db-structure/src/parser/sql/postgresql/processSQLInChunks.test.ts:158]: Test case is easy to understand 👍🏻

**Review by MH4GF — APPROVED** [review]
LGTM 👍🏻 
It looks fine!

File: .changeset/cyan-crabs-kiss.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+"@liam-hq/db-structure": patch
+"@liam-hq/erd-core": patch
```
Comment (MH4GF) [inline .changeset/cyan-crabs-kiss.md:3]: Is this necessary?

```suggestion
```

Comment (hoshinotsuyoshi) [reply .changeset/cyan-crabs-kiss.md:3]: sorry, it is unnecessary.

**Comment by hoshinotsuyoshi** [comment]
Thanks, I'll merge now
